src/utils/extractAmendedBillableTestResults.ts

```diff
@@ -3,60 +3,34 @@ import { Differences, DifferencesEntries } from './differences';
 import { TestResultModel, TestType } from './testResult';
 
 export const extractAmendedBillableTestResults = (currentRecord: TestResultModel, previousRecord: TestResultModel) => {
-  const testTypeValuesToCheck = ['testCode'] as const;
-  const testResultValuesToAdd = ['vin', 'vrm'] as const;
-  const testResultValuesToCheck = ['testStationPNumber', ...testResultValuesToAdd] as const;
-
-  type TestTypesValues = typeof testTypeValuesToCheck[number];
-  type TestResultValues = typeof testResultValuesToCheck[number] | typeof testResultValuesToAdd[number];
+  const testTypeValues = ['testCode'] as const;
+  const testResultValues = ['testStationPNumber', 'vin', 'vrm'] as const;
 
   const fieldsChanged: Differences[] = [];
-  currentRecord.testTypes.forEach((testType) => {
+  currentRecord.testTypes.forEach((currentTestType) => {
     const fields: DifferencesEntries[] = [];
 
-    testTypeValuesToCheck.forEach((key) => {
-      const oldTestType: TestType = previousRecord.testTypes.find(
-        (previousTestType) => previousTestType.testNumber === testType.testNumber,
-      );
-      if (oldTestType[key] !== testType[key]) {
-        fields.push({
-          fieldName: key,
-          oldValue: oldTestType[key as TestTypesValues],
-          newValue: testType[key as TestTypesValues],
-        });
-      }
-    });
+    const previousTestType: TestType = previousRecord.testTypes.find(
+      (testType) => testType.testNumber === currentTestType.testNumber,
+    );
 
-    testResultValuesToCheck.forEach((key) => {
-      if (currentRecord[key] !== previousRecord[key]) {
-        fields.push({
-          fieldName: key,
-          oldValue: previousRecord[key as TestResultValues],
-          newValue: currentRecord[key as TestResultValues],
-        });
-      }
-    });
+    const hasAnyFieldChanged = testResultValues.some((field) => currentRecord[field] !== previousRecord[field])
```

Comment:
You could possibly use the `.every` instead of `.some` with the inverse of the condition on this line to return all the fields that _have_ changed and then your subsequent forEach takes less steps?

Reply:
Oh wait are they wanting all the fields for the test result if it's changed?